src/main/connector/system/LinuxService/LinuxService.yaml

```diff
@@ -0,0 +1,62 @@
+extends:
+- ../System/System
+metrics:
+  system.service.status:
+    description: Service Status
+    type:
+      stateSet:
+        - running
+        - dead
+        - exited
+connector:
+  displayName: LinuxService
+  platform: Any platform with LinuxOS
+  reliesOn: Linux OsCommands
+  information: Gives OS specific service information and metrics
+  detection:
+    connectionTypes:
+    - remote
+    - local
+    appliesTo:
+    - linux
+    criteria:
+    - type: commandLine
+      commandLine: /usr/bin/uname -o || /bin/uname -o
+      expectedResult: GNU/Linux
+      errorMessage: Not a valid Linux host.
+    tags: [ system ]
+monitors:
+  service:
+    simple:
+      type: multiInstance
+      sources:
+        serviceInfo:
+          # serviceName;load;active;sub;description
+          type: commandLine
+          commandLine: /usr/bin/systemctl list-units --type=service --all |  grep -E ${var::serviceNames} | sed 's/^[^a-zA-Z0-9]*//'

```

Comment:
For the sed, this was because of a special character found with systemctl '●':
![image](https://github.com/user-attachments/assets/caa2528c-7fc9-4f52-8194-431795d94dbc)

For some unfathomable reason, no amount of AWK manipulations was able to deal with said character (I had a long discussion with Nassim on this a few months back). So we ended up adding a sed specifically to deal with this.

Reply:
Since this is actually only for "not-found" services, we could also just exclude them.

Reply:
The special char is a terminal control char to print in red color. Other similar characters may be found in the output. To disable colors, use this environment variable:

```
      $SYSTEMD_COLORS
           Takes a boolean argument. When true, systemd and related utilities will use colors in their output,
           otherwise the output will be monochrome. Additionally, the variable can take one of the following special
           values: "16", "256" to restrict the use of colors to the base 16 or 256 ANSI colors, respectively. This
           can be specified to override the automatic decision based on $TERM and what the console is connected to.
```

The command will therefore be;

```bash
SYSTEMD_COLORS=false ; /usr/bin/systemctl list-units --type=service --all
```

To be tested!

Reply:
In this case, we can simply remove these non-alphanumeric chars in the AWK script with:

```awk
{ sub(/^[^a-zA-Z0-9]+/, "");
```

The YAML becomes then:

```yaml
          computes:
          - type: awk
            script: |
              /\.service/ {
                sub(/^[^a-zA-Z0-9]+/, "");
                sub("\.service", "");
                printf($1 ";" $2 ";" $3 ";" $4 ";")
                for (i=5; i<=NF; i++) {
                  printf("%s ", $i)
                }
                printf("\n")
              }
```